app/models/miq_queue.rb

```diff
@@ -623,6 +623,15 @@ def self.display_name(number = 1)
 
   private
 
+  # NOTE: this will lookup an unknown zone every time.
+  def validate_zone_name
+    unless zone.nil? || self.class.valid_zone_names[zone] ||= Zone.in_my_region.exists?(:name => zone)
```

Comment:
I'm finding this construct very hard to mentally parse with `unless`, `||` and `||=` in the same line.

Reply:
Also I think you are caching the zone object itself as the hash key as opposed to the zone name or zone id. Name can be problematic if a zone is renamed, but id would work.

Reply:
I will rework.
Yea, it is the name. and yes, it is strange